numpy/distutils/system_info.py

```diff
@@ -498,17 +505,46 @@ def calc_libraries_info(self):
                 dict_append(info, **i)
             else:
                 log.info('Library %s was not found. Ignoring' % (lib))
+            i = self.check_libs(r_dirs, [lib])
+            if i is not None:
+                # Swap library keywords found to runtime_library_dirs
+                # the libraries are insisting on the user having defined
+                # them using the library_dirs, and not necessarily by
+                # runtime_library_dirs
+                del i['libraries']
+                i['runtime_library_dirs'] = i.pop('library_dirs')
+                dict_append(info, **i)
```

Comment:
Then all those extra keys would already be updated by this code above right?

            i = self.check_libs(dirs, [lib])
            if i is not None:
                dict_append(info, **i)